jujugui/static/gui/src/app/components/deployment/choose-cloud/choose-cloud.js

```diff
@@ -22,18 +22,25 @@ YUI.add('deployment-choose-cloud', function() {
 
   juju.components.DeploymentChooseCloud = React.createClass({
 
-    // XXX huwshimi 22 March 2016: this should be replaced with a passed in prop
-    // once the data is available.
-    CREDENTIALS: [{
-      id: 'my-cloud-credentials',
-      cloud: 'aws',
-      title: 'My cloud credentials',
-      owner: 'Me!',
-      uses: 3
-    }],
-
     propTypes: {
-      changeState: React.PropTypes.func.isRequired
+      changeState: React.PropTypes.func.isRequired,
+      jem: React.PropTypes.object.isRequired
+    },
+
+    getInitialState: function() {
+      return {
+        credentials: []
+      };
+    },
+
+    componentWillMount: function() {
+      this.props.jem.listTemplates((error, data) => {
+        if (error) {
+          console.error('Unable to list templates', error);
```

Comment:
Agreed. I already have a card to polish the UI on this after we finish the "create template" work, which would be a good time to turn this into a user-facing error message. Comment added.